### ISSUE TITLE
Repurpose `GET /session/properties` to do something useful

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#1892]
This API call is made on every single page. It returns `(setting/public-settings)`, which is the exact same thing that goes into `METABASE_BOOTSTRAP` in every page. Effectively we are making the user wait for an extra API round trip for data we already have.

**Comment by agilliland** [comment]
That is not true and the endpoint does serve a purpose.  We use it to refresh settings which can be changed at runtime because not all of the properties in MetabaseBootstrap are completely static.

There is definitely room for optimization though and it's true that right when the application loads we probably don't need to call this endpoint.

**Comment by camsaul** [comment]
I'm not saying it's not *theoretically* useful but from what I can see the only time this endpoint gets called at all is when the application loads. So right now, not actually useful.

**Comment by agilliland** [comment]
Look at the function `refreshSiteSettings` in services.js

**Comment by camsaul** [comment]
So... it gets called *after* someone updates a setting in the admin panel? To fetch the changes they just made? The only way this can possibly return data we don't already have is if two people are changing settings simultaneously.

This API call *could* be construed as useful if it instead happened *when the Settings page is visited*. Then we could fetch changes that may have occurred since the initial load.

But it's not. Luckily, `GET /setting/` *is* called when the page is visited. It returns settings that can change dynamically -- the things we actually need. `GET /session/properties` returns the same things, plus a bunch of extra static ones that won't change.

**TL;DR** `GET /session/properties` is unnecessary and never called in a place that would make it actually useful.

**Comment by camsaul** [comment]
As discussed between @tlrobinson and I it would be nice to use this endpoint to pull things out of MetabaseBootstrap that aren't needed for setup so we can only return them when a user is authenticated. Additionally, this endpoint could return settings only admins can see if the user is an admin.

**Comment by tlrobinson** [comment]
Further discussed with @camsaul:

We should:

1. add `:public` and `:authenticated` properties to `defsetting`
2. replace the manual `(defn public-settings ...)` function in `public_settings.clj` with one that just returns the settings where `:public true` is set (we'll need to add some computed settings)
3. `GET /api/session/properties` should return settings where `:authenticated true` (and `:public true`?) is set
4. change some of the `:public` settings that don't need to be public to be `:authenticated`.

We could also eliminate `GET /api/session/properties` and just return the `:authenticated` settings in `GET /api/user/current` which is already called (twice for some reason!) when a user loads the app or logs in.

On the frontend we should change all settings to use their canonical `lisp-case` names so they stay in sync if an admin changes a setting. We should also use the Redux store as the source of truth for settings values, and have `metabase/lib/settings.js` get settings values from there.

**Comment by camsaul** [comment]
Should be easy enough.

Other notes: 

*  Use computed settings (magic getters) for the stuff like `features` that comes back as a map in the EE settings

**Comment by camsaul** [comment]
Closing this as implemented by #11239
